### PR TITLE
Add download artifacts in engage publication

### PR DESCRIPTION
**Pull request by jduehring**
This is a seperate PR to only include the backend from #2769.

All download artifacts are now automatically included in the engange-publication channel: 
The publish-engage WOH now has an additional configuration key list-elements which defaults to true and adds the additional information. If no additional information is preferred, the key must explicitly be set to false.

### Your pull request should…

* [ ] have a concise title
* [ ] [close an accompanying issue](https://help.github.com/en/articles/closing-issues-using-keywords) if one exists
* [ ] [be against the correct branch](https://docs.opencast.org/develop/developer/development-process#acceptance-criteria-for-patches-in-different-versions)
* [ ] include migration scripts and documentation, if appropriate
* [ ] pass automated tests
* [ ] have a clean commit history
* [ ] [have proper commit messages (title and body) for all commits](https://medium.com/@steveamaza/e028865e5791)

### CODE REVIEW

**Review by gregorydlogan — APPROVED** [review]

**Review by mtneug — CHANGES_REQUESTED** [review]
A republish (strategy `merge`) with only ACLs / metadata seams to remove everything else.

File: docs/guides/admin/docs/workflowoperationhandlers/publish-engage-woh.md [hunk 1]
```diff
@@ -28,9 +28,9 @@ Parameter Table
 |streaming-source-tags      |Specifies which media should be published to the streaming server                            |
 |streaming-target-tags      |Add tags (comma separated) to published media                                                |
 |streaming-target-subflavors|Subflavor to use for distributed material                                                    |
-|merge-force-flavors        |Flavors of elements for which an update is enforced when merging catalogs.                  |
+|merge-force-flavors        |Flavors of elements for which an update is enforced when merging catalogs.                   |
 |                           |Defaults to `dublincore/*,security/*`.
-
+|list-elements              |Include all media package elements in the publication. Defaults to `true`.                   |
```
Comment (mtneug) [inline docs/guides/admin/docs/workflowoperationhandlers/publish-engage-woh.md]: The consensus during the dev meeting seems to be to always enable listing the published elements. So we don't need this option.

**Review by mtneug — CHANGES_REQUESTED** [review]
How does this interact with the `retract-partial` WOH?

**Review by mtneug — CHANGES_REQUESTED** [review]
The following options are not considered at the moment:

* `download-target-subflavors`
* `download-target-tags`
* `streaming-target-subflavors`
* `streaming-target-tags`

`getMediaPackageForSearchIndex` deals with these options. So this should be generalized for all mediapackage variants.

**Comment by mtneug** [comment]
> A republish (strategy `merge`) with only ACLs / metadata seams to remove everything else.

Actually, don't bother. `merge` is very broken in general (i.e. also in `publish-configure`) and conventionally thinking about this problem, a proper `merge` implementation seems very hard.

**Comment by gregorydlogan** [comment]
@mtneug This is waiting on your previous comment, correct?

File: docs/guides/admin/docs/workflowoperationhandlers/publish-engage-woh.md (hunk 1 above)
Comment (jduehring) [reply docs/guides/admin/docs/workflowoperationhandlers/publish-engage-woh.md]: Sorry, I lost track of this PR. Removed the option to disable listing the published elements. 

It seems that the `retract-partial` WOH is currently not able to find the added elements. I need to check that...

**Comment by jduehring** [comment]
> The following options are not considered at the moment:
> 
> * `download-target-subflavors`
> * `download-target-tags`
> * `streaming-target-subflavors`
> * `streaming-target-tags`
> 
> `getMediaPackageForSearchIndex` deals with these options. So this should be generalized for all mediapackage variants.

So if I understand correctly, it would be sufficient to select the new elements, after all target tags/flavors got applied in `getMediaPackageForSearchIndex`? I adjusted the code to make sure of that.

File: docs/guides/admin/docs/workflowoperationhandlers/publish-engage-woh.md (hunk 1 above)
Comment (jduehring) [reply docs/guides/admin/docs/workflowoperationhandlers/publish-engage-woh.md]: Actually it was a user error on my side...tested partialy retracting some published elements and everything seems to work as expected.

**Comment by gregorydlogan** [comment]
@mtneug @lkiesow are you happy with this?  It's (currently) the last 10.x pr.

**Comment by lkiesow** [comment]
Looks good and works as expected → merging

```json
…
      "channel": "engage-player",
      "id": "ee90b59a-5550-40b5-9d31-1fa32f1f0f57",
      "media": [
        {
          "has_audio": true,
          "framerate": 50,
          "description": "",
          "bitrate": 9390780,
          "url": "http://localhost:8080/static/mh_default_org/engage-player/c0b5b00f-b217-48c6-ae3a-8cb4c03382b6/3d36f118-597b-4ead-b382-16e24cc0916d/goat.mp4",
          "has_video": true,
          "tags": [
            "atom",
            "default",
            "engage-download",
            "engage-streaming",
            "rss"
          ],
          "flavor": "presenter/preview",
          "duration": 4160,
          "size": 4955801,
          "framecount": 208,
          "checksum": "02780f257e9397883517ebed78a2df98 (md5)",
          "width": 1920,
          "id": "e1c554d8-2a47-4759-a5e1-c6d4d6c4153a",
          "mediatype": "video/mp4",
          "height": 1080
        }
      ],
…
```

**Pull request merged**